File: 12. Integer to Roman/solution.py
class Solution(object):
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        st=""
        while num!=0:
            if num>=1000:
                st+="M"
                num-=1000
            elif num>=900:
                st+="CM"
                num-=900
            elif num>=500:
                st+="D"
                num-=500
            elif num>=400:
                st+="CD"
                num-=400
            elif num>=100:
                st+="C"
                num-=100
            elif num>=90:
                st+="XC"
                num-=90
            elif num>=50:
                st+="L"
                num-=50
            elif num>=40:
                st+="XL"
                num-=40
            elif num>=10:
                st+="X"
                num-=10
            elif num>=9:
                st+="IX"
                num-=9
            elif num>=5:
                st+="V"
                num-=5
            elif num>=4:
                st+="IV"
                num-=4
            elif num>=1:
                st+="I"
                num-=1
        return st

File: 12. Integer to Roman/Solution.java
class Solution {
    public String intToRoman(int num) {
        String roman = "";
        while (num>0){
            if (num>=1000){
                roman=roman+"M";
                num-=1000;
            }else if (num>=900){
                roman=roman+"CM";
                num-=900;
            }else if (num>=500){
                roman=roman+"D";
                num-=500;
            }else if (num>=400){
                roman=roman+"CD";
                num-=400;
            }else if (num>=100){
                roman=roman+"C";
                num-=100;
            }else if (num>=90){
                roman=roman+"XC";
                num-=90;
            }else if (num>=50){
                roman=roman+"L";
                num-=50;
            }else if (num>=40){
                roman=roman+"XL";
                num-=40;
            }else if (num>=10){
                roman=roman+"X";
                num-=10;
            }else if (num>=9){
                roman=roman+"IX";
                num-=9;
            }else if (num>=5){
                roman=roman+"V";
                num-=5;
            }else if (num>=4){
                roman=roman+"IV";
                num-=4;
            }else if (num>=1){
                roman=roman+"I";
                num-=1;
            }
        }
        return roman;
    }
}